contracts/pause/src/lib.sw

```diff
@@ -0,0 +1,52 @@
+library pause;
+
+dep interface;
+dep r#storage;
+
+use std::{logging::log, storage::{get, store}};
+
+use interface::{PausedEvent, UnpausedEvent};
+use storage::PAUSED_STORAGE_KEY;
+
+enum PausedState {
+    Unpaused: (),
+    Paused: (),
+}
+
+#[storage(read)]
+pub fn require_unpaused() {
+    require(!is_paused(), "contract is paused");
+}
+
+/// Returns whether the contract is paused.
+#[storage(read)]
+pub fn is_paused() -> bool {
+    match get_pause_state() {
+        PausedState::Unpaused => false,
+        PausedState::Paused => true,
+    }
+}
+
+/// Gets the PausedState from storage.
+#[storage(read)]
+fn get_pause_state() -> PausedState {
+    get(PAUSED_STORAGE_KEY).unwrap_or(PausedState::Unpaused)
```

Comment:
this default is a little scary to me

Reply:
the alternative is an explicit initializer, wdyt?

imo this is well tested so it's pretty much the same as in Solidity where the `paused` storage variable would be implicitly initialized to its zero value, which means that it's not paused